libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -56,6 +61,23 @@ export class Subspace {
     return this._projects;
   }
 
+  public getPnpmOptions(): PnpmOptionsConfiguration | undefined {
+    if (!this._pnpmOptions) {
+      if (FileSystem.exists(`${this.getSubspaceConfigFolder()}/${RushConstants.pnpmConfigFilename}`)) {
```

Comment:
If the file doesn't exist, this function will check for existence every time it's called. Consider noting a not-exist state for the cached value.

We do that in other places. Poke around for examples. I'm pretty sure some things initialize the cached value to an "unknown" state and use `undefined` as a nonexistent file.

Reply:
Do you know where I can find an example of this behavior? I've tried looking around for an "unknown" state but I can't seem to find anything other than using unknown as a type

Reply:
I think he means something like this, where `false` represents cached knowledge that the item is missing.

https://github.com/microsoft/rushstack/blob/641f8f8aed4f1cde81bdc2dca9f915e83e0c326a/libraries/rush-lib/src/api/RushProjectConfiguration.ts#L199

The `Map<>` class actually distinguishes the difference between `map.get(key) === undefined` versus `map.has(key) === false` which could also be used to represent this distinction.  However, an explicit value such as `false` is more readable. And probably more efficient, since ECMAScript inexplicably neglected to provide a single API that can test both states (such as `bool Dictionary.TryGetValue(key, out value)` from .NET).

Reply:
Fixed in 6a452438b9db6aec3c514b8fad6098310672a6a8.

Since there is no `Map` in this case, the logic is much cleaner to just clarify the name `_cachedPnpmOptions` and add a second variable `_cachedPnpmOptionsInitialized` to track whether it is initialized.

I also tried combining the states (`PnpmOptionsConfiguration | undefined | false`), but the result is awkward:
- If `false` means "the file is missing", then we counterintuitively have to do  `return this._cachedPnpmOptions === false ? undefined : this._cachedPnpmOptions `   **-OR-**
- If `false` means "cache not initialized yet", then `undefined` means "the file is missing" which is different semantics from every other variable of this type.